molr-mole-core/src/main/java/io/molr/mole/core/support/domain/MissionStub2.java

```diff
@@ -1,4 +1,30 @@
 package io.molr.mole.core.support.domain;
 
-public interface MissionStub2<P1, P2, R> {
+import io.molr.commons.domain.Mission;
+import io.molr.commons.domain.Placeholder;
+
+import java.util.Map;
+
+public class MissionStub2<P1, P2, R> {
+
+    protected StubData data;
+
+    MissionStub2(Mission mission, Placeholder<R> returnType, Placeholder<P1> p1, Placeholder<P2> p2) {
+        data = StubData.from(mission)
+                .setReturnType(returnType)
+                .addParameter(p1)
+                .addParameter(p2);
```

Comment:
In the end, I think your StubData is a good conceopt (something generic below the strict interface). However, Consider to make even the StubData an immutable object and having an internal builder. The call here would e.g. look like this:

```
data = StubData.from(mission)
                .setReturnType(returnType)
                .addParameter(p1)
                .addParameter(p2)
                .build();
```